test/src/main/java/org/axonframework/test/fixture/AxonTestPhase.java

```diff
@@ -361,16 +369,12 @@ interface Command extends Message<Command> {
             Command success();
 
             /**
-             * Expect the last command handler from the When phase to return a result message that matches the given
-             * {@code matcher}.
-             * <p>
-             * Only take commands into account that were dispatched explicitly with the {@link When#command}. Hence, do
-             * not take into accounts commands dispatched as side effects of the message handlers.
+             * Allow to consume the result of command message which have been published during the "when" phase.
              *
-             * @param matcher The matcher to verify the actual result message against.
+             * @param consumer Consumes the command result. You may place your own assertions here.
              * @return The current Then instance, for fluent interfacing.
              */
-            Command resultMessage(@Nonnull Matcher<? super CommandResultMessage<?>> matcher);
+            Command resultMessage(@Nonnull Consumer<? super CommandResultMessage<?>> consumer);
```

Comment:
I am not 100% sure about this signature.

I think I would like to include `asserting` in the method name, to make it clear that the user has to make its own assertions. Such as `assertingResultMessage`, or maybe `resultMessageMatches`

Reply:
Of course, this pointer would go for the other methods

Reply:
I see you did `eventsMatch` for events, because it takes a `Predicate`. Something similar, but for assert consumers.

We could also maybe make a functional interface, so it shows in the signature: `resultMessage(AssertionFunction<CommandResultMessage<?>>`

Reply:
Ahhh, sorry for the quadruple ocmment, but I just recalled assertJ uses `satisfy` for this:

So `resultMessageSatifies(AssertionFunction<CommandResultMessage<?>>)`. What do you think?

Reply:
I was also thinking about that.
I prefer to have just `resultMessage`, but as you said I've done it because of the predicate. 
But maybe it'd better to rename all assertion methods like:
`eventsSatisfy(Consumer)`
`eventsMatch(Predicate)`
`resultMessageSatifies(Consumer)`
`resultMessageMatch(Predicate)`

WDYT?

Reply:
`resultMessageSatifies(Consumer<CommandResultMessage<?>>)` is enough? Or do you think there are some advantages of dedicated interface?

Reply:
Ough, it works without postfixes... Hmm... maybe the wildcard was a problem.

Reply:
I've renamed and keep argumnt types. Please check one more time.